packages/fast-jss-manager-react/src/manage-jss.tsx

```diff
@@ -109,44 +81,11 @@ function manageJss<S, C>(styles?: ComponentStyles<S, C>): <T>(Component: React.C
             constructor(props: T) {
                 super(props);
 
-                // On construction, check if the style object or component object have already been used.
-                // If not, we need to store them in our weakmaps for later use
-                if (!Boolean(JSSManager.styleMap.get(styles))) {
-                    JSSManager.styleMap.set(styles, uniqueId());
-                }
-
-                if (!Boolean(JSSManager.componentMap.get(Component))) {
-                    JSSManager.componentMap.set(Component, uniqueId());
-                }
-
-                const styleId: string = JSSManager.styleMap.get(styles);
-                const componentId: string = JSSManager.componentMap.get(Component);
-
-                // Store the instance id as a combination of the generated IDs
-                this.instanceId = `${componentId}${styleId}`;
-                let separatedStylesInstance: ISeparatedStylesheet<S, C> = JSSManager.separatedStyles[this.instanceId];
-
-                // Check if we have a separatedStyles object stored at the instanceId key.
-                // If we don"t, we need to create that object
-                if (!Boolean(separatedStylesInstance)) {
-                    separatedStylesInstance = JSSManager.separatedStyles[this.instanceId] = this.separateStyles(styles);
-                }
+                JSSManager.styles = styles;
 
-                // Now lets store those newly created stylesheet objects in state so we can easily reference them later
-                // Since dynamic styles can be different across components, we should create the dynamic styles as a
-                // new object so that identity checks between dynamic stylesheets do not pass.
                 const state: IJSSManagerState = {};
 
-                // Extract the static stylesheet and dynamic style object and apply them to the state
-                // object if they exist
-                const staticStyleSheet: StyleSheet = separatedStylesInstance.staticStyleSheet;
-                const dynamicStyles: StyleSheet = separatedStylesInstance.dynamicStyles;
-
-                if (Boolean(staticStyleSheet)) {
-                    state.staticStyleSheet = separatedStylesInstance.staticStyleSheet;
-                }
-
-                if (Boolean(dynamicStyles)) {
+                if (Boolean(JSSManager.styles)) {
```

Comment:
we should probably just rename to `styleSheet` - no reason to differentiate these as "dynamic" anymore